### PR TITLE
port katello_sync to apypie

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
@Manisha15 @mdellweg @ehelms @sean797 plz review :)

File: plugins/modules/katello_sync.py [hunk 1]
```diff
@@ -95,52 +95,38 @@
...
-    except Exception as e:
-        module.fail_json(msg=e)
+    params['organization'] = module.find_resource_by_name('organizations', params['organization'], thin=True)
+    scope = {'organization_id': params['organization']['id']}
+    params['product'] = module.find_resource_by_name('products', params['product'], params=scope, thin=True)
+    if 'repository' in params:
+        product_scope = {'product_id': params['product']['id']}
+        params['repository'] = module.find_resource_by_name('repositories', params['repository'], params=product_scope, thin=True)
+        changed, task = module.resource_action('repositories', 'sync', {'id': params['repository']['id']})
+        if params['synchronous']:
```
Comment (evgeni) [inline plugins/modules/katello_sync.py]: this might become easier after we merge #405

Comment (mdellweg) [reply plugins/modules/katello_sync.py]: ... which is the case now.

Comment (evgeni) [reply plugins/modules/katello_sync.py]: just pushed an updated version

**Pull request merged**